near-abi-client/tests/mod.rs

```diff
@@ -0,0 +1,42 @@
+use near_abi_client::Generator;
+use quote::quote;
+use std::fs;
+use tempdir::TempDir;
+
+#[test]
+fn test_generate_abi() -> anyhow::Result<()> {
+    let tmp_dir = TempDir::new("adder-generated-code")?;
+    let tmp_dir_path = tmp_dir.into_path();
+    Generator::new(tmp_dir_path.clone())
+        .file("tests/adder.json")
+        .generate()?;
+
+    let generated_code = fs::read_to_string(tmp_dir_path.join("adder.rs"))?;
+    let expected = quote! {
+        pub type Pair = Vec<i64>;
+        pub struct ExtAbi {
+            pub contract: workspaces::Contract,
+        }
+        impl ExtAbi {
+            pub async fn add(
+                &self,
+                worker: &workspaces::Worker<impl workspaces::Network>,
+                a: Pair,
+                b: Pair
```

Comment:
What do you think about only requiring references or something bounded on `AsRef<T>`? Definitely tradeoffs and unclear what is a good balance of efficiency and ergonomics, but for example, here there _shouldn't_ be a reason why a vec would need to be allocated here.

Nothing to change, just curious if you've thought about this

Reply:
I have entertained this thought, and to me it seemed like it would not be that straightforward. Naively pretending `&` would result in weird types like `&String`, `&Vec<T>`, `&Option<T>` etc. Should I make an exception for certain types? If so, how do I know that `Pair` is actually a type alias for `Vec<T>`? This also raises the question of the possibility of generating structs that do not need to own the data (e.g., an error struct with a static message string). Should they hold `Cow<'a, str>` and `Cow<'a, [u8]>` or just `&str` and `&[u8]`? There is a lot of ambiguity that I honestly did not want to deal with at this point, so decided to go with the most straightforward way of everything owning all the data for now.

Reply:
yeah, sounds good, let's leave as this for now and we can always migrate the APIs later if we find a clean way to make it more ergonomic/optimized